clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```
@@ -177,6 +181,19 @@ public void closeHandlers() throws IOException {
         channel.close();
     }
 
+    @Override
+    public void reopen(File f) {
+        if(OperatingSystem.IS_WINDOWS){
+            try {
+                channel = openChannel(f, true, true, this.start, false);
+            }
+            catch (IOException e){
```

Please don't just swallow the exception. Can we log the exception in case we hit this. Also, what would happen if we don't catch the exception and just let it go, meaning have throws IOException to the method